test/stream_router_test.exs

```diff
@@ -0,0 +1,113 @@
+defmodule TestSink do
+  defstruct pid: nil, forward_to: nil, val: nil
+  use StreamRouter
```

Comment:
Because this injects a `handle_info/2` clause to match on `$gen_route` we get warnings about non-adjacent function clauses. Would be nice if there was a good way to handle this other than moving the `use StreamRouter` call to above the other ordinary handle_info clauses